### PR TITLE
Feature/bulk lead delete

### PR DESCRIPTION
**Pull request by annshress**
Bulk deleting leads

Endpoints:
```js
POST 
/api/v1/project/<PROJECT-ID>/leads/dry-bulk-delete/ 
data: {leads: <LIST of lead ids>}
response: number of entities that will cascade delete (for now: entries, assessments)

POST
/api/v1/project/<PROJECT-ID>/leads/bulk-delete/ 
data: {leads: <LIST of lead ids>}
response: 204 No Content
```

### CODE REVIEW

**Comment by AdityaKhatri** [comment]
@annshress , lets use array instead of comma-separated ids, also let's rename ids to leads.
Let's scope the URL by project Id as well.

**Comment by thenav56** [comment]
> @annshress , lets use array instead of comma-separated ids, also let's rename ids to leads.
> Let's scope the URL by project Id as well.

Yeah, scoping by project will make the permission check easier too.

File: deep/permissions.py [hunk 1]
```diff
@@ -65,6 +65,46 @@ def has_permission(self, request, view):
         return projects_count == len(project_ids)
 
 
+class DeleteLeadPermission(permissions.BasePermission):
+    """Checks if user can delete lead(s)"""
+    def has_permission(self, request, view):
+        if request.method not in ('DELETE',):
+            return True
+
+        project_id = request.parser_context['kwargs']['project']
```
Comment (annshress) [inline deep/permissions.py]: @bewakes  Can you verify if this permission handler is okay?

**Review by frozenhelium — APPROVED** [review]

File: deep/permissions.py (hunk 1 above)
Comment (bewakes) [reply deep/permissions.py]: this seems okay. But just to be on the safe side, use `.get('project')`.

Comment (bewakes) [reply deep/permissions.py]: However, I do not understand(or maybe if I forgot if it was there already) why we are passing project id. Shouldn't leads be sufficient to get project information?

Comment (tnagorra) [reply deep/permissions.py]: I think we are limiting a user's action to a specific project first.
We shouldn't be able to bulk delete leads of different projects. @thenav56 right?

**Pull request merged**